### PR TITLE
feat: Add forge and output oracle

### PR DESCRIPTION
**Pull request by tynes**
**Description**
replaces https://github.com/ethereum-optimism/optimism/pull/2466

This is a PR on a new feature branch to be used for migrating the L2OutputOracle contract into the monorepo. 

It makes changes to some of the config, such as the contracts package.json and relevant ignore files.
As for contracts, it does not yet touch any of the pre-existing ones.

### CODE REVIEW

**Comment by maurelian** [comment]
This should build on [feat/bridge-via-output-oracle](https://github.com/ethereum-optimism/optimism/tree/feat/bridge-via-output-oracle) which is currently the same as develop

**Review by maurelian — APPROVED** [review]
Best PR ever. 👍

File: packages/contracts/.solhintignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ node_modules
 # WETH9 is a standard we should not modify it.
 contracts/L2/predeploys/WETH9.sol
 contracts/L2/predeploys/IWETH9.sol
+lib
```
Comment (maurelian) [inline packages/contracts/.solhintignore:6]: Sorry, I did this.

File: packages/contracts/package.json [hunk 2]
```diff
@@ -17,13 +17,18 @@
     "standards"
   ],
   "scripts": {
+    "postinstall": "yarn submodules || exit 0",
+    "prebuild": "yarn submodules || exit 0",
```
Comment (maurelian) [inline packages/contracts/package.json:21]: Do we need both of these?

File: packages/contracts/slither.config.json [hunk 3]
```diff
@@ -8,5 +8,5 @@
   "hardhat_ignore_compile": false,
   "disable_color": false,
   "exclude_dependencies": false,
-  "filter_paths": "@openzeppelin|hardhat|contracts/test-helpers|contracts/test-libraries|contracts/L2/predeploys/WETH9.sol"
+  "filter_paths": "@openzeppelin|hardhat|contracts/test-helpers|contracts/test-libraries|contracts/L2/predeploys/WETH9.sol|lib|test"
```
Comment (maurelian) [inline packages/contracts/slither.config.json:11]: Ahh, that's the right way to do this!

File: packages/contracts/.solhintignore (hunk 1 above)
Comment (maurelian) [reply packages/contracts/.solhintignore:6]: (no newline that is)

**Comment by mslipper** [comment]
@mergifyio refresh

File: packages/contracts/package.json (hunk 2 above)
Comment (tynes) [reply packages/contracts/package.json:21]: We could probably get away with just 1 of them

**Pull request merged**